src/main/scala/com/microsoft/ml/spark/logging/BasicLogging.scala

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.logging
+
+import org.apache.spark.internal.Logging
+import spray.json.{DefaultJsonProtocol, RootJsonFormat}
+
+case class BasicLogInfo(
+                       uid: String,
+                       className: String,
+                       method: String
+                       )
+
+object LogJsonProtocol extends DefaultJsonProtocol {
+  implicit val LogFormat: RootJsonFormat[BasicLogInfo] = jsonFormat3(BasicLogInfo)
+}
+
+import LogJsonProtocol._
+import spray.json._
+
+trait BasicLogging extends Logging {
+
+  val uid: String
+
+  protected def logBase(methodName: String): Unit = {
+    logInfo("metrics/ " + BasicLogInfo(uid, getClass.toString, methodName).toJson.toString())
```

Comment:
```suggestion
    logInfo("metrics/ " + BasicLogInfo(uid, getClass.toString, methodName).toJson.compactPrint())
```
not sure if toString is properly supported in all cases. this is the explicit method that stringifyies in spray

Reply:
I tried compactPrint previously and I'm not sure what's the issue that causes the logs can't be properly output to Kusto, in Kusto table they show as 'metrics/ {"[REDACTED]"}'.. So I'm trying to see if toString will work.

Reply:
Figured out that this has nothing to do with compactPrint, changed it back. That issue is related to log4j properties setup.